### DISCUSSION TITLE
Sharing context/state with multiple screens

### DISCUSSION THREAD

**Discussion by AliAhmadi74** [wix/react-native-navigation#7627] [Q&A] [answered]
hi i have question.
I started digging around in the source code, and found out that the store property for <RecoilRoot> is actually available, but not exported So what could potentially work is just a HoC function which exports a Root bound to the shared store and can therefore be used to create multiple screens (= roots) with the same store.
but I saw that the contextProvider not rerender all screen. making it entirely incompatible with react-native-navigation.
any help?

**Comment by sssajjad007** [comment, marked as answer]
Re-renders aren’t triggered across screens when the context changes,
cuz contexts does not support proxy any more (hermes not supporting Proxy). i have same problem and i decided to use Redux instead.
like this

    return Navigation.registerComponent(
      screenName,
      () => props =>
        (
          <ReduxProvider store={store}>
                <Component {...props} />
          </ReduxProvider>
        ),
      () => Component, // for navigation options
    );
  });
